tests/gnmi/helper.py

```diff
@@ -1,16 +1,44 @@
 import time
 import re
+from tests.common.utilities import get_image_type
 
-GNMI_CONTAINER_NAME = 'telemetry'
-GNMI_PROGRAM_NAME = 'telemetry'
-GNMI_CONFIG_KEY = 'TELEMETRY|gnmi'
+GNMI_CONTAINER_NAME = ''
+GNMI_PROGRAM_NAME = ''
+GNMI_PORT = 0
 # Wait 15 seconds after starting GNMI server
 GNMI_SERVER_START_WAIT_TIME = 15
 
 
+def gnmi_container(duthost):
+    global GNMI_CONTAINER_NAME
+    if len(GNMI_CONTAINER_NAME) == 0:
+        if get_image_type(duthost) == "public":
+            GNMI_CONTAINER_NAME = "telemetry"
+        else:
+            GNMI_CONTAINER_NAME = "gnmi"
+    return GNMI_CONTAINER_NAME
+
+
+def gnmi_program(duthost):
+    global GNMI_PROGRAM_NAME
+    if len(GNMI_PROGRAM_NAME) == 0:
+        if get_image_type(duthost) == "public":
+            GNMI_PROGRAM_NAME = "telemetry"
+        else:
+            GNMI_PROGRAM_NAME = "gnmi-native"
+    return GNMI_PROGRAM_NAME
+
+
 def gnmi_port(duthost):
-    port = duthost.shell("redis-cli -n 4 hget '%s' 'port'" % GNMI_CONFIG_KEY)['stdout']
-    return port
+    global GNMI_PORT
+    if GNMI_PORT == 0:
+        if get_image_type(duthost) == "public":
+            GNMI_CONFIG_KEY = 'TELEMETRY|gnmi'
+        else:
+            GNMI_CONFIG_KEY = 'GNMI|gnmi'
+        port = duthost.shell("redis-cli -n 4 hget '%s' 'port'" % GNMI_CONFIG_KEY)['stdout']
```

Comment:
>redis-cli -n 4 [](http://example.com/codeflow?start=30&length=14)

`sonic-db-cli CONFIG_DB` will be better.

Reply:
Updated.